homework9/hw9task3/iterators_3/universal_file_counter.py

```diff
@@ -0,0 +1,32 @@
+"""
+Write a function that takes directory path, a file extension and an optional tokenizer.
+It will count lines in all files with that extension if there are no tokenizer.
+If a the tokenizer is not none, it will count tokens.
+For dir with two files from hw1.py:
+#>>> universal_file_counter(test_dir, "txt")
+6
+#>>> universal_file_counter(test_dir, "txt", str.split)
+6
+"""
+import os
+from os.path import getsize
+from pathlib import Path
+from typing import Callable, Optional
+
+
+def universal_file_counter(
+    dir_path: Path, file_extension: str, tokenizer: Optional[Callable] = None
+) -> int:
+    amount_of_tokens_or_lines = 0
+    all_files = [file for file in os.listdir(dir_path) if file.endswith(file_extension)]
+    for file_name in all_files:
+        file_path = str(dir_path) + "/" + file_name
+        if getsize(file_path) != 0:
```

Comment:
Why do we need this check?

Reply:
Just to skip this file if it is empty, no need to open it and take time

Reply:
I suppose it is a correct check because I faced with the following problem: if we split text by some symbol (for example `data.split('c')`) and in case file is emty, so, our amount_of_tokens will increase in 1, e.g.
''.split('c') will add +1.  But in case split() has no argument (split by space), nothinf will change with amount of tokens. So, split() works differently depends on argument presence. Is it a correct work of split()?

Reply:
What operation you do with if and else branch. What they have in common?

Reply:
In common: we incease amount_of_tokens_or_lines. You mean to do like `amount_of_tokens_or_lines += len(tokenizer(all_data)) if tokenizer else all_data.count("\n")` ?

Reply:
This is a step forward.  What else we are doing here?

Reply:
Check if file is not empty, then read data in file and do manipulations with data - common

Reply:
Okey, probably I guessed what you mean. Check, please.

Reply:
You guess it properly.  But could we use it as the default value? What do requirements say about it?

Reply:
I suppose no, okey, tokenizer should be None as a default value

Reply:
I expected something like:

```python
if tokenizer is None:
   tokenizer = lambda data: data.split("\n")
```